vuln-reach/src/javascript/lang/accesses.rs

```diff
@@ -188,17 +194,28 @@ impl<'a> AccessGraph<'a> {
 
                     if lhs.byte_range().contains(&node.start_byte()) {
                         // Find next parent identifier if node is in LHS of the assignment.
-                        return Self::find_accessor(cursor_cache, parent);
+                        cur_depth = depth;
+                        node = parent;
```

Comment:
Node is never actually used for anything other than constructing the cursor I think? So it should be fine removing this statement too?

It kinda makes sense since this loop inherently moves through the parents so a no-op is basically the same as recursion (except for `class_declaration` with `depth == 0`).

Reply:
I did only reassign depth at first, but then I noticed that `&node.start_byte()` is actually used, right on top of this snippet. So a proper recursion would entail reassigning the node as well. 

I'm not super clear on where of those branches we want to have a proper recursion/"reinitialization"; one could argue that just keeping going with the loop all the way through could be ok too. Could we maybe make a special case of anonymous declarations without using `depth`?